src/canister_tests/src/tests.rs

```diff
@@ -955,6 +956,262 @@ mod http_tests {
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+
+        let metrics = flows::get_metrics(&env, canister_id);
+
+        let (min_user_number, _) =
+            framework::parse_metric(&metrics, "internet_identity_min_user_number");
+        let (max_user_number, _) =
+            framework::parse_metric(&metrics, "internet_identity_max_user_number");
+        assert_eq!(min_user_number, 10_000);
+        assert_eq!(max_user_number, 3_784_872);
+        Ok(())
+    }
+
+    /// Verifies that the user count metric is updated correctly.
+    #[test]
+    fn metrics_user_count_should_increase_after_register() -> Result<(), CallError> {
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+
+        for count in 0..2 {
+            let metrics = flows::get_metrics(&env, canister_id);
+
+            let (user_count, _) = framework::parse_metric(&metrics, "internet_identity_user_count");
+            assert_eq!(user_count, count);
+
+            flows::register_anchor(&env, canister_id);
+        }
+        Ok(())
+    }
+
+    /// Verifies that the signature count metric is updated correctly.
+    #[test]
+    fn metrics_signature_count_should_increase_after_prepare_delegation() -> Result<(), CallError> {
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM.clone());
+        let frontend_hostname = "https://some-dapp.com";
+        let user_number = flows::register_anchor(&env, canister_id);
+
+        for count in 0..3 {
+            let metrics = flows::get_metrics(&env, canister_id);
+            let (signature_count, _) =
+                framework::parse_metric(&metrics, "internet_identity_signature_count");
+            assert_eq!(signature_count, count);
+
+            api::prepare_delegation(
+                &env,
+                canister_id,
+                principal_1(),
+                user_number,
+                frontend_hostname.to_string(),
+                ByteBuf::from(format!("session key {}", count)),
+                None,
+            )?;
+        }
+
+        // long after expiry (we don't want this test to break, if we change the default delegation expiration)
+        env.advance_time(Duration::from_secs(365 * 24 * 60 * 60));
+        // preparing a new delegation prunes old ones
+        api::prepare_delegation(
+            &env,
+            canister_id,
+            principal_1(),
+            user_number,
+            frontend_hostname.to_string(),
+            ByteBuf::from("last session key"),
+            None,
+        )?;
+
+        let metrics = flows::get_metrics(&env, canister_id);
+        let (signature_count, _) =
+            framework::parse_metric(&metrics, "internet_identity_signature_count");
+        assert_eq!(signature_count, 1); // old ones pruned and a new one created
```

Comment:
nit: how about checking each individual? i.e. first `advance_time` to prune, check metric, then `prepare_delegation`, check metric

Reply:
`advance_time` is not enough, we also need to do an update call that prunes expired signatures. There following update calls prune signatures: `add`, `create_challenge`, `prepare_delegation`, `register`, `remove`

I chose to call `prepare_delegation` again, because this is the least likely one of those calls to be changed to stop pruning signatures (as it is the only one that actually executes code related to signatures).

I changed the comment a little bit to make it more clear.